File: app/src/main/java/net/bloople/stories/StoryParser.java
package net.bloople.stories;

import java.io.IOException;
import java.io.Reader;
import java.util.Scanner;

class StoryParser {
    private Scanner scanner;

    StoryParser(Reader reader) {
        scanner = new Scanner(reader);
        scanner.useDelimiter("(?:\r?\n[\u200B\uFEFF]*){2,}+");
    }

    boolean hasNext() throws IOException {
        return scanner.hasNext();
    }

    String next() throws IOException {
        return scanner.next();
    }
}

File: app/src/main/java/net/bloople/stories/StoryParser.kt
package net.bloople.stories

import java.io.IOException
import java.io.Reader
import java.util.*
import kotlin.Throws

internal class StoryParser(reader: Reader) {
    private val scanner: Scanner

    @Throws(IOException::class)
    operator fun hasNext(): Boolean {
        return scanner.hasNext()
    }

    @Throws(IOException::class)
    operator fun next(): String {
        return scanner.next()
    }

    init {
        scanner = Scanner(reader)
        scanner.useDelimiter("(?:\r?\n[\u200B\uFEFF]*){2,}+")
    }
}